metadatastore/core.py

```diff
@@ -1043,7 +1044,7 @@ def find_events(start_col, start_cache,
         # wrap it our fancy dict
         ev = doc.Document('Event', ev)
         yield ev
-
+    events.close()
```

Comment:
Ideally they would have a context manager on find so you could do

```python
with foo.find(...) as events:
    for ev in events:
        yield ev
```

but I could not find that in the API quickly.